### PR TITLE
refactor(compiler-cli): Add signals to internal directive metadata

### PR DESCRIPTION
**Pull request by atscott**
This commit adds the `signals: boolean` property to the internal directive/component metadata. This does not add it to the public API yet, as the feature has no internal support other than compiler detection.

### CODE REVIEW

**Review by devversion — APPROVED** [review]
LGTM. I had a similar change in my prototyping. Just one minor suggestion

**Review by jessicajaniuk — APPROVED** [review]
reviewed-for: fw-core, public-api

File: packages/core/src/render3/component.ts [hunk 1]
```diff
@@ -120,6 +120,8 @@ export interface ComponentMirror<C> {
    * Note: an extra flag, not present in `ComponentFactory`.
    */
   get isStandalone(): boolean;
+  /** @internal */
```
[the remark this answers is not in the document]
Comment (atscott) [reply packages/core/src/render3/component.ts]: I think I'll keep the `@internal` but also add the TODO. For the annotations, I think it doesn't matter much. I could see an argument for completely removing it from the mirror so the getter doesn't even exist, but it probably doesn't matter that much here either.

**Review by dylhunn — APPROVED** [review]
reviewed-for: public-api

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

Reviewed-for: public-api 
Reviewed-for: fw-core

**Review by devversion — APPROVED** [review]

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit 5214df49586f57f28618556423c906ef5f607e2e.

**Pull request closed without merging**